addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -182,21 +182,30 @@ export const editorCommands = {
         // In case the html inserted is all contained in a single root <p> or <li>
         // tag, we take the all content of the <p> or <li> and avoid inserting the
         // <p> or <li>. The same is true for a <pre> inside a <pre>.
+        const isBlockElement = (node) => isBlock(node) && !["TABLE", "UL", "OL", "DIV"].includes(node.nodeName) && startNode.textContent !== "";
```

Comment:
@Zynton As per you request I have cherrypicked this commit from https://github.com/odoo/odoo/pull/126375/
But I would like to change this variable to `isContextBlock` and move the `startNode.textContent !== ""` to if conditions below rather. Waiting for you confirmation.

Reply:
For later reference, [here](https://github.com/odoo/odoo/pull/115019#discussion_r1469428577) is the comment where I suggested that. Is the spec something that was validated by @pierre-pde and/or @dmo-odoo though?

Since it's a commit from a canceled task, I'd say feel free to modify it if needed (and put yourself as co-author of the commit). I think your suggestion to split it is good but I don't understand the name since this checks if it's a block and if it's _not_ a table/list/div. Also, in the editor we usually use single quotes so you can change these double quotes while you're at it.

Reply:
Yes, I was the one to create the task in the first place and it was in response to multiple people reporting the issue. For the record, I'm ok with the split of the condition.